### PR TITLE
feat(route): add ELSEVIER journal

### PR DESCRIPTION
**Pull request by Derekmini**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/elsevier/signal-processing/latest
/elsevier/signal-processing/vol/192
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [x] Documentation
  - [x] CN
  - [x] EN
- [x] 全文获取 fulltext
  - [x] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] 日期和时间 date and time
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/elsevier/latest.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    const resp = await got('https://www.sciencedirect.com');
+    const cookie = resp.headers['set-cookie'].join(' ');
+    const journal = ctx.params.journal;
+    const hostUrl = 'https://www.sciencedirect.com/journal/' + journal;
+
+    const hostResponse = await got({
+        method: 'get',
+        url: hostUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
```
Comment (TonyRL) [inline lib/v2/elsevier/latest.js:16]: Why specified `www.sciencedirect.com` again in `Host` headers when `www.sciencedirect.com` is already specified in `hostUrl`?

File: lib/v2/elsevier/latest.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        volume = 'Volume ' + ltstUrl.match('vol/(.*)/suppl')[1];
+    } else if (ltstUrl.match('issue') !== null) {
+        volume = 'Volume ' + ltstUrl.match('vol/(.*)/issue')[1] + ' Issue ' + ltstUrl.match('/issue/(.*)')[1];
+    }
+
+    const ltstResponse = await got({
+        method: 'get',
+        url: ltstUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
```
Comment (TonyRL) [inline lib/v2/elsevier/latest.js:40]: Why specified `www.sciencedirect.com` again in `Host` headers when `www.sciencedirect.com` is already specified in `ltstUrl`?

File: lib/v2/elsevier/latest.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        });
+
+    await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const itemResponse = await got({
+                    method: 'get',
+                    url: 'https://www.sciencedirect.com/science/article/pii/' + item.id,
+                    headers: {
+                        Host: 'www.sciencedirect.com',
```
Comment (TonyRL) [inline lib/v2/elsevier/latest.js:80]: Why specified `www.sciencedirect.com` again in `Host` headers when `www.sciencedirect.com` is already specified in `url`?

File: lib/v2/elsevier/volume.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    const cookie = resp.headers['set-cookie'].join(' ');
+    const journal = ctx.params.journal;
+    const volume = 'Volume ' + ctx.params.id.replace('-', ' Issue ');
+    const volUrl = 'https://www.sciencedirect.com/journal/' + journal + '/vol/' + ctx.params.id.replace('-', '/issue/');
+
+    const volResponse = await got({
+        method: 'get',
+        url: volUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
```
Comment (TonyRL) [inline lib/v2/elsevier/volume.js:17]: Why specified `www.sciencedirect.com` again in `Host` headers when `www.sciencedirect.com` is already specified in `volUrl`?

File: lib/v2/elsevier/volume.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        });
+
+    await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const itemResponse = await got({
+                    method: 'get',
+                    url: 'https://www.sciencedirect.com/science/article/pii/' + item.id,
+                    headers: {
+                        Host: 'www.sciencedirect.com',
```
Comment (TonyRL) [inline lib/v2/elsevier/volume.js:58]: Why specified `www.sciencedirect.com` again in `Host` headers when `www.sciencedirect.com` is already specified in `url`?

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/elsevier/latest.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    const cookie = resp.headers['set-cookie'].join(' ');
+    const journal = ctx.params.journal;
+    const hostUrl = 'https://www.sciencedirect.com/journal/' + journal;
+
+    const hostResponse = await got({
+        method: 'get',
+        url: hostUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36',
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/elsevier/latest.js:17]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/latest.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const itemResponse = await got({
+                    method: 'get',
+                    url: 'https://www.sciencedirect.com/science/article/pii/' + item.id,
+                    headers: {
+                        Host: 'www.sciencedirect.com',
+                        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36',
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/elsevier/latest.js:81]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/volume.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    const journal = ctx.params.journal;
+    const volume = 'Volume ' + ctx.params.id.replace('-', ' Issue ');
+    const volUrl = 'https://www.sciencedirect.com/journal/' + journal + '/vol/' + ctx.params.id.replace('-', '/issue/');
+
+    const volResponse = await got({
+        method: 'get',
+        url: volUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36',
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/elsevier/volume.js:18]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/volume.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const itemResponse = await got({
+                    method: 'get',
+                    url: 'https://www.sciencedirect.com/science/article/pii/' + item.id,
+                    headers: {
+                        Host: 'www.sciencedirect.com',
+                        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36',
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/elsevier/volume.js:59]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/latest.js (hunk 1 above)
Comment (Derekmini) [reply lib/v2/elsevier/latest.js:16]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/latest.js (hunk 2 above)
Comment (Derekmini) [reply lib/v2/elsevier/latest.js:40]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/latest.js (hunk 3 above)
Comment (Derekmini) [reply lib/v2/elsevier/latest.js:80]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/volume.js (hunk 4 above)
Comment (Derekmini) [reply lib/v2/elsevier/volume.js:17]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/volume.js (hunk 5 above)
Comment (Derekmini) [reply lib/v2/elsevier/volume.js:58]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: lib/v2/elsevier/latest.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    } else if (ltstUrl.match('issue') !== null) {
+        volume = 'Volume ' + ltstUrl.match('vol/(.*)/issue')[1] + ' Issue ' + ltstUrl.match('/issue/(.*)')[1];
+    }
+
+    const ltstResponse = await got({
+        method: 'get',
+        url: ltstUrl,
+        headers: {
+            Host: 'www.sciencedirect.com',
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36',
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/elsevier/latest.js]: Thanks your attention, I have modified it.
```javascript
const config = require('@/config').value;
const headers = {
    Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    Referer: 'https://id.elsevier.com/',
    Host: host.replace('https://', ''),
    'User-Agent': config.ua,
    cookie,
};
```

File: docs/journal.md [hunk 11]
```diff
@@ -80,6 +80,20 @@ pageClass: routes
...
+<Route author="Derekmini sunsunwolf-swb" example="/elsevier/signal-processing/latest" path="/elsevier/:journal/latest" :paramsDesc="['期刊名称，复制 URL 中 tocSection 部分']" radar="1" rssbud="1">
+
+</Route>
+
+### 指定卷中的文章
+
+<Route author="Derekmini sunsunwolf-swb" example="/elsevier/signal-processing/vol/192" path="/elsevier/:journal/vol/:id" :paramsDesc="['期刊名称，复制 URL 中 tocSection 部分','卷号 (如果 `Issue` 存在, 使用 `Volume-Issue`, e.g., `/elsevier/aace-clinical-case-reports/vol/7-6`)']" radar="1" rssbud="1">
+
+</Route>
+
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply docs/journal.md:96]: Thanks your attention, I have added en docs.

File: docs/journal.md [hunk 12]
```diff
@@ -80,6 +80,20 @@ pageClass: routes
...
+### 最新文章
+
+<Route author="Derekmini sunsunwolf-swb" example="/elsevier/signal-processing/latest" path="/elsevier/:journal/latest" :paramsDesc="['期刊名称，复制 URL 中 tocSection 部分']" radar="1" rssbud="1">
+
+</Route>
+
+### 指定卷中的文章
+
+<Route author="Derekmini sunsunwolf-swb" example="/elsevier/signal-processing/vol/192" path="/elsevier/:journal/vol/:id" :paramsDesc="['期刊名称，复制 URL 中 tocSection 部分','卷号 (如果 `Issue` 存在, 使用 `Volume-Issue`, e.g., `/elsevier/aace-clinical-case-reports/vol/7-6`)']" radar="1" rssbud="1">
```
Comment (TonyRL) [inline docs/journal.md:93]:
```suggestion
<Route author="Derekmini sunsunwolf-swb" example="/elsevier/signal-processing/vol/192" path="/elsevier/:journal/vol/:id" :paramsDesc="['期刊名称，复制 URL 中 tocSection 部分','卷号 (如果 `Issue` 存在，使用 `Volume-Issue`，e.g., `/elsevier/aace-clinical-case-reports/vol/7-6`)']" radar="1" rssbud="1">
```

Comment (TonyRL) [reply docs/journal.md:93]: Please fix them in another PR.

Comment (Derekmini) [reply docs/journal.md:93]: Thanks for your attention! I have modified it and will get a new pr.

**Pull request merged**